### PR TITLE
travis: publish docs

### PR DESCRIPTION
**Pull request by Itxaka**
Publish docs when merged into the master branch to github pages

### CODE REVIEW

**Comment by Itxaka** [comment]
This *needs* a token from a user with access to the repo as it pushes to the `gh-pages` branch. The token should be created by a bot/automated user and added to travis as `token` var

Not sure who to talk to or who is in charge of the bots

**Comment by Itxaka** [comment]
Also, it works really nice: https://itxaka.github.io/socok8s/deploy-guide/index.html

File: .travis.yml [hunk 1]
```diff
@@ -2,12 +2,27 @@ language: python
...
+      stage: tests
       script: find . -path "./submodules" -prune -o -type f -name "*.sh" -print | xargs -n1 shellcheck -x --format=gcc
     - name: "playbook syntax"
+      stage: tests
       script: ansible-lint playbooks/*.yml
+    - name: "build and publish docs"
+      stage: publish docs
+      install:
+        - pip install tox
+      script: tox
```
Comment (evrardjp) [inline .travis.yml]: can you change this to tox -e docs ?
Tox might grow with reno and linting.

File: .travis.yml [hunk 2]
```diff
@@ -2,12 +2,27 @@ language: python
 cache: pip
 dist: xenial
 
-matrix:
```
Comment (evrardjp) [inline .travis.yml]: Could you explain why this movesfrom matrix to jobs?

File: .travis.yml (hunk 1 above)
Comment (Itxaka) [reply .travis.yml]: :+1:

File: .travis.yml (hunk 2 above)
Comment (evrardjp) [reply .travis.yml]: I meant in the commit message, not here :)

Comment (Itxaka) [reply .travis.yml]: To move this into different stages to we dont build the docs if there is a failure on the tests, mostly it gains us time and we can reuse it later for extra steps that we dont necessarily want to run unless the linting is passed first (like unit tests that take longer? molecule testing?)

Kind of like the jenkins pipeline, if step X fails do not move into the next one.

Comment (Itxaka) [reply .travis.yml]: oh, that makes sense....

**Comment by evrardjp** [comment]
@Itxaka still don't merge?

**Comment by Itxaka** [comment]
> @Itxaka still don't merge?

yup yup, need to find out who is the bot owner for github to obtain a token and put it on the repo secret vars in travis so it can push to github pages. Will try to do it today as its currently using my own token, which is bad :)

**Comment by toabctl** [comment]
@Itxaka any update on this?

**Comment by Itxaka** [comment]
> @Itxaka any update on this?

@toabctl unfortunately nope, I could not get the owner of the susegithubapi account to appear :/

I guess we can move this forward and use one of my tokens meanwhile? That usually leads to forgetting about it until it breaks thought :D

**Comment by Itxaka** [comment]
This will be moved into jenkins instead

**Pull request closed without merging**